src/diskquota_utility.c

```diff
@@ -1294,17 +1294,25 @@ worker_spi_get_extension_version(int *major, int *minor)
  * Get the list of oids of the tables which diskquota
  * needs to care about in the database.
  * Firstly the all the table oids which relkind is 'r'
- * or 'm' and not system table.
+ * or 'm' and not system table. On init stage, oids from
+ * diskquota.table_size are added to invalidate them.
  * Then, fetch the indexes of those tables.
  */
 
 List *
-get_rel_oid_list(void)
+get_rel_oid_list(bool is_init)
 {
 	List *oidlist = NIL;
 	int   ret;
 
-	ret = SPI_execute_with_args("select oid from pg_class where oid >= $1 and (relkind='r' or relkind='m')", 1,
+#define SELECT_FROM_PG_CATALOG_PG_CLASS \
+	"select oid from pg_catalog.pg_class where oid >= $1 and (relkind='r' or relkind='m')"
+#define SELECT_FROM_DISKQUOTA_TABLE_SIZE "select tableid from diskquota.table_size where segid = -1"
```

Comment:
The SELECT_FROM_DISKQUOTA_TABLE_SIZE macro is used only once. So let's remove it and write
```
" union distinct"
" select tableid from diskquota.table_size where segid = -1"
```

Reply:
applied